packages/manager/apps/telecom/src/app/telecom/pack/orderFollowUp/orderFollowUp.controller.js

```diff
@@ -0,0 +1,83 @@
+import isEmpty from 'lodash/isEmpty';
+import last from 'lodash/last';
+
+export default class OrderFollowUpCtrl {
+  /* @ngInject */
+  constructor(
+    $stateParams,
+    $translate,
+    $http,
+    TucToast,
+    TucToastError,
+    ORDER_STATUS,
+  ) {
+    this.$http = $http;
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+    this.ORDER_STATUS = ORDER_STATUS;
+  }
+
+  getByPackName(packName) {
+    const url = `/pack/xdsl/${packName}/orderFollowUp`;
+
+    return this.$http
+      .get(url)
+      .then((response) => {
+        return response.data;
+      })
+      .catch((err) => {
+        throw new this.TucToastError(err);
+      });
+  }
+
+  loadData() {
+    this.getByPackName(this.packName)
+      .then((data) => {
+        this.events = data.map((elt) => {
+          this.allSuccess = data.every((item) => item.status === 'done');
+
+          const comments = elt.comments
+            ? elt.comments
+                .map((thisComment) => `<span>${thisComment}</span>`)
+                .join('')
+            : '';
+
+          return {
+            badgeClass: this.ORDER_STATUS[elt.status]?.class,
+            badgeIconClass: this.ORDER_STATUS[elt.status]?.icon,
+            name: elt.name,
+            status: elt.status,
+            when: elt.doneDate,
+            contentHtml: comments,
+            side: 'right',
+          };
+        });
+
+        this.orderStatus = last(data)?.status;
+        this.loading = false;
+      })
+      .catch((err) => {
+        this.events = [];
+        this.loading = false;
+        this.TucToastError(err);
+      });
+  }
+
+  $onInit() {
+    this.events = [];
+    this.loading = true;
+    this.allSuccess = false;
+    this.packName = this.$stateParams.packName;
+
+    if (isEmpty(this.packName)) {
+      this.TucToast.error(
+        this.$translate.instant('xdsl_order_follow_up_total_error'),
+      );
+      this.loading = false;
+    }
+
+    this.loadData();
+  }
```

Comment:
For the code easier to read, I suggest putting the $onInit at the beginning just after the constructor